controllers/ptpinterface_controller.go

```diff
@@ -674,3 +691,84 @@ func (r *PtpInterfaceReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&starlingxv1.PtpInterface{}).
 		Complete(r)
 }
+
+// Verify whether we have annotation restore-in-progress
+func (r *PtpInterfaceReconciler) checkRestoreInProgress(instance *starlingxv1.PtpInterface) bool {
+	restoreInProgress, ok := instance.Annotations[cloudManager.RestoreInProgress]
+	if ok && restoreInProgress != "" {
+		return true
+	}
+	return false
+}
+
+// Update status and clear annotation
+func (r *PtpInterfaceReconciler) RestorePtpinterfaceStatus(instance *starlingxv1.PtpInterface) error {
+	annotation := instance.GetObjectMeta().GetAnnotations()
+	config, ok := annotation[cloudManager.RestoreInProgress]
+	if ok {
+		restoreStatus := &cloudManager.RestoreStatus{}
+		err := json.Unmarshal([]byte(config), &restoreStatus)
+		if err == nil {
+			if restoreStatus.InSync != nil {
+				instance.Status.InSync = *restoreStatus.InSync
+			}
+			if restoreStatus.DeploymentScope != nil {
+				if strings.ToLower(*restoreStatus.DeploymentScope) == "bootstrap" ||
+					strings.ToLower(*restoreStatus.DeploymentScope) == "principal" {
+					instance.Status.DeploymentScope = *restoreStatus.DeploymentScope
+				} else {
+					r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "Invalid deploymentScope value '%s', scope will be set to boostrap", *restoreStatus.DeploymentScope)
+					instance.Status.DeploymentScope = "bootstrap"
+				}
+			} else {
+				r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "The 'deploymentScope' is not specified, assuming 'bootstrap' scope.")
+				instance.Status.DeploymentScope = "bootstrap"
+			}
+			if restoreStatus.StrategyRequired != nil {
```

Comment:
Regarding the strategy required status, just update all of them to not_required

Reply:
Changes are done